sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java

```diff
@@ -199,11 +199,20 @@ public Context getSharedSpanBuilder(String spanName, Context context) {
      */
     @Override
     public void addEvent(String eventName, Map<String, Object> traceEventAttributes, OffsetDateTime timestamp) {
+        addEvent(eventName, traceEventAttributes, timestamp, Context.NONE);
```

Comment:
Why we need this api , Context.NONE will not have the span info ?

Reply:
Since this is an existing API and we do not want to break any existing users using this API by removing it.

Reply:
Can we do deprecation,document update because this method will return without doing anything, so could be misleading.

Reply:
BTW, it doesn't look like there are any users https://github.com/Azure/azure-sdk-for-java/search?q=addevent. are we expecting anyone outside of this repo to use it?

I understand it's a public API and can't be simply removed. Agree that deprecation would be great. 
Just curious if we expect anyone else to ever use it :)

Reply:
>are we expecting anyone outside of this repo to use it?

Anyone using their own implementation of the azure-core, Tracer class would/could be using this API.
Also, we sort to deprecation as the last resort and if use any workarounds instead according to our guidance here
https://github.com/Azure/azure-sdk-for-java/wiki/Deprecation

Reply:
yeah, I'm checking my understanding: if something calls into addEvent (which would be one of the Azure SDKs), then custom Tracer implementation would override it. But nothing calls it yet, correct?

Reply:
is this needed to maintain existing behavior? (agree hopefully no one is using it anyways)
```suggestion
        addEvent(eventName, traceEventAttributes, timestamp, new Context(PARENT_SPAN_KEY, Span.current()));
```

Reply:
Agree with @trask, I think it would make sense to keep this overload with a defaulting behavior to look for the current span in scope. For the users, who do not want any specific span for the event to be attached to. 
@lmolkova thoughts?

Reply:
(I would also support deprecation, this method does seem a bit out of place since the azure tracing API seems generally designed around passing explicit context)